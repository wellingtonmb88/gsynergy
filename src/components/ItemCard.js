import React, { Component } from "react";
import { StyleSheet, Text, View, Image } from "react-native";
import PropTypes from "prop-types";
import ShimmerPlaceHolder from "react-native-shimmer-placeholder";

export default class itemCard extends Component {
  static propTypes = {
    item: PropTypes.object.isRequired
  };

  state = {
    imageLoaded: false
  };

  render() {
    const { item } = this.props;
    const { imageLoaded } = this.state;
    const imageStyle = imageLoaded ? styles.image : styles.imageMin;

    return (
      <View style={styles.container}>
        <ShimmerPlaceHolder
          style={styles.shimmerImageComponent}
          autoRun={true}
          visible={imageLoaded}
          backgroundColorBehindBorder={"white"}
        />
        <Image
          style={imageStyle}
          source={{ uri: item.image_url }}
          onLoad={() => {
            this.setState({ imageLoaded: true });
          }}
        />
        <ShimmerPlaceHolder
          style={styles.shimmerTextComponent}
          autoRun={true}
          visible={imageLoaded}
        >
          <Text style={styles.text}>{item.text}</Text>
        </ShimmerPlaceHolder>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    height: 200
  },
  shimmerImageComponent: {
    height: 180,
    borderRadius: 14
  },
  image: {
    height: 180,
    overflow: "hidden",
    borderRadius: 14
  },
  imageMin: {
    width: 1,
    height: 1
  },
  shimmerTextComponent: {
    marginTop: 12,
    marginLeft: 9
  },
  text: {
    textAlign: "left",
    color: "#5C5C5C",
    marginTop: 12,
    marginLeft: 9,
    fontFamily: "Helvetica Neue",
    fontSize: 18
  }
});
