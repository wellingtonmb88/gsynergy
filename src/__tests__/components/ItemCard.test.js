import React from "react";
import ItemCard from "../../components/ItemCard";
import { shallow } from "enzyme";

const props = {
  item: {
    id: "1",
    text: "name1",
    image_url: "https://unsplash.it/1000/1000"
  }
};

describe("render the component ItemCard", () => {
  beforeAll(() => {
    jest.useFakeTimers();
  });

  it("renders with item", () => {
    const tree = shallow(<ItemCard {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
