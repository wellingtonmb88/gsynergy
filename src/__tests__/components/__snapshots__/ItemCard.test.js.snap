// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the component ItemCard renders with item 1`] = `
<Component
  style={
    Object {
      "height": 200,
    }
  }
>
  <ShimmerPlaceHolder
    autoRun={true}
    backgroundColorBehindBorder="white"
    colorShimmer={
      Array [
        "#ebebeb",
        "#c5c5c5",
        "#ebebeb",
      ]
    }
    duration={1000}
    hasBorder={false}
    height={15}
    location={
      Array [
        0.3,
        0.5,
        0.7,
      ]
    }
    reverse={false}
    style={
      Object {
        "borderRadius": 14,
        "height": 180,
      }
    }
    visible={false}
    width={200}
    widthShimmer={0.7}
  />
  <Image
    onLoad={[Function]}
    source={
      Object {
        "uri": "https://unsplash.it/1000/1000",
      }
    }
    style={
      Object {
        "height": 1,
        "width": 1,
      }
    }
  />
  <ShimmerPlaceHolder
    autoRun={true}
    backgroundColorBehindBorder="white"
    colorShimmer={
      Array [
        "#ebebeb",
        "#c5c5c5",
        "#ebebeb",
      ]
    }
    duration={1000}
    hasBorder={false}
    height={15}
    location={
      Array [
        0.3,
        0.5,
        0.7,
      ]
    }
    reverse={false}
    style={
      Object {
        "marginLeft": 9,
        "marginTop": 12,
      }
    }
    visible={false}
    width={200}
    widthShimmer={0.7}
  >
    <Component
      style={
        Object {
          "color": "#5C5C5C",
          "fontFamily": "Helvetica Neue",
          "fontSize": 18,
          "marginLeft": 9,
          "marginTop": 12,
          "textAlign": "left",
        }
      }
    >
      name1
    </Component>
  </ShimmerPlaceHolder>
</Component>
`;
